pyatlan/samples/events/lambda_scorer.py

```diff
@@ -0,0 +1,244 @@
+from pyatlan.model.enums import (
+    AtlanCustomAttributePrimitiveType,
+    BadgeComparisonOperator,
+    BadgeConditionColor,
+    CertificateStatus,
+)
+from pyatlan.model.events import AtlanEvent, AtlanEventPayload
+from pyatlan.model.structs import BadgeCondition
+from pyatlan.model.typedef import AttributeDef, CustomMetadataDef
+
+CM_DAAP = "DaaP"
+CM_ATTR_DAAP_SCORE = "Score"
+
+SCORED_ATTRS = [
+    "description",
+    "userDescription",
+    "ownerUsers",
+    "ownerGroups",
+    "meanings",
+    "__hasLineage",
+    "classifications",
+    "inputToProcesses",
+    "outputFromProcesses",
+    "assignedEntities",
+    "seeAlso",
+    "links",
+    "certificateStatus",
+    "readme",
+]
+client = AtlanClient()
+
+
+def _create_cm_if_not_exists() -> Optional[str]:
+    """
+    Creates the custom metadata structure, if it does not already exist. (If it
+    already exists, does nothing.)
+
+    :returns: unique identifier for the custom metadata structure
+    """
+
+    try:
+        return CustomMetadataCache.get_id_for_name(CM_DAAP)
+    except NotFoundError:
+        try:
+            cm_def = CustomMetadataDef.create(display_name=CM_DAAP)
+            cm_def.attribute_defs = [
+                AttributeDef.create(
+                    display_name=CM_ATTR_DAAP_SCORE,
+                    attribute_type=AtlanCustomAttributePrimitiveType.DECIMAL,
+                )
+            ]
+            cm_def.options = CustomMetadataDef.Options.with_logo_as_emoji("🔖")
+            client.create_typedef(cm_def)
+            print("Created DaaP custom metadata structure.")
+            badge = Badge.create(
+                name=CM_ATTR_DAAP_SCORE,
+                cm_name=CM_DAAP,
+                cm_attribute=CM_ATTR_DAAP_SCORE,
+                badge_conditions=[
+                    BadgeCondition.create(
+                        badge_condition_operator=BadgeComparisonOperator.GTE,
+                        badge_condition_value="75",
+                        badge_condition_colorhex=BadgeConditionColor.GREEN,
+                    ),
+                    BadgeCondition.create(
+                        badge_condition_operator=BadgeComparisonOperator.LT,
+                        badge_condition_value="75",
+                        badge_condition_colorhex=BadgeConditionColor.YELLOW,
+                    ),
+                    BadgeCondition.create(
+                        badge_condition_operator=BadgeComparisonOperator.LTE,
+                        badge_condition_value="25",
+                        badge_condition_colorhex=BadgeConditionColor.RED,
+                    ),
+                ],
+            )
+            try:
+                client.asset.save(badge)
+                print("Created DaaP completeness score badge.")
+            except Exception:
+                print("Unable to create badge over DaaP score.")
+            return CustomMetadataCache.get_id_for_name(CM_DAAP)
```

Comment:
I confused because if an exception occurs in the save you will still get here but the get_id_for_name will fail won't it?

Reply:
I think you are correct! The thing that we can do is we can only attempt to retrieve the ID from CustomMetadataCache if the badge is successfully saved so moving `return CustomMetadataCache.get_id_for_name(CM_DAAP)` inside the try block